src/notes/alias.entity.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import {
+  Column,
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  Unique,
+} from 'typeorm';
+import { Note } from './note.entity';
+import { PrimaryValueTransformer } from './primary.value-transformer';
+
+@Entity()
+@Unique('Only one primary alias per note', ['note', 'primary'])
+export class Alias {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  /**
+   * the actual alias
+   */
+  @Column({
+    nullable: false,
+    unique: true,
+  })
+  name: string;
+
+  /**
+   * Is this alias the primary alias, by which people access the note?
+   */
+  @Column({
+    comment:
+      'This field telly you if this is the primary alias of the note. If this field is null, that means false.',
```

Comment:
Why should this field be nullable?

Reply:
Because of this line:

https://github.com/hedgedoc/hedgedoc/blob/4be02939b3525c1c43c80f2f999558e613841f7a/src/notes/alias.entity.ts#L17

If a non-primary alias would be saved with `primary: false,` you could only every have one non-primary and one primary alias. But a nullable field does not have such problems. It's really a bit ugly, but this way the DB keeps track one note really only has one primary alias and I think this worth the slightly uglier code. I'll try to add a better explanation in the code. Maybe not in the DB comment though…

Reply:
Yes, I understand it's a bit hacky but should be ok.